storm-core/src/jvm/org/apache/storm/metric/ClusterConsumerExecutor.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.metric;
+
+import com.google.common.util.concurrent.MoreExecutors;
+import org.apache.storm.Config;
+import org.apache.storm.metric.api.DataPoint;
+import org.apache.storm.metric.api.IClusterMetricsConsumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+public class ClusterConsumerExecutor {
+    public static final Logger LOG = LoggerFactory.getLogger(ClusterConsumerExecutor.class);
+
+    private IClusterMetricsConsumer metricsConsumer;
+    private String consumerClassName;
+    private Object registerationArgument;
+    private ExecutorService consumerExecutorService;
+
+    public ClusterConsumerExecutor(String consumerClassName, Object registerationArgument) {
+        this.consumerClassName = consumerClassName;
+        this.registerationArgument = registerationArgument;
+    }
+
+    public void prepare() {
+        try {
+            metricsConsumer = (IClusterMetricsConsumer)Class.forName(consumerClassName).newInstance();
```

Comment:
Yes right.
Users are encouraged to package their consumer as a fat jar and place to system environment STORM_EXT_CLASSPATH_DAEMON or /extlib-daemon.
Placing jar to system environment STORM_EXT_CLASSPATH or /extlib also works but it will affect non-daemon (including worker) so former would be better way.